honeybee_schema/radiance/properties.py

```diff
@@ -0,0 +1,173 @@
+        description='A string for a Honeybee Radiance Modifier.'
+        )
+
+    modifier_blk: str = Field(
+        default=None,
+        min_length=1,
+        max_length=100,
+        description='A string for a Honeybee Radiance Modifier to be used '
+                    'in direct solar simulations and in isolation studies (assessing'
+                    'the contribution of individual objects).'
+        )
+
+
+class _PropertiesBase(IDdBaseModel):
+    """Base class of Radiance Properties."""
+
+    modifier: ModifierBase = Field(
+        default=None,
+        description='A Honeybee Radiance Modifier for the object.'
+        )
+
+    modifier_blk: ModifierBase = Field(
+        default=None,
+        description='A Honeybee Radiance Modifier to be used for this object'
+                    'in direct solar simulations and in isolation studies (assessing'
+                    'the contribution of individual objects).'
+        )
+
+
+class ApertureRadiancePropertiesAbridged(_PropertiesBase):
+    """Radiance Properties for Honeybee Aperture Abridged."""
+
+    type: constr(regex='^ApertureRadiancePropertiesAbridged$') = \
+        'ApertureRadiancePropertiesAbridged'
+
+
+class ApertureRadianceProperties(_PropertiesBase):
+    """Radiance Properties for Honeybee Aperture."""
+
+    type: constr(regex='^ApertureRadianceProperties$') = 'ApertureRadianceProperties'
+
+
+class DoorRadiancePropertiesAbridged(_PropertiesBaseAbridged):
+    """Radiance Properties for Honeybee Door Abridged."""
+
+    type: constr(regex='^DoorRadiancePropertiesAbridged$') = \
+        'DoorRadiancePropertiesAbridged'
+
+
+class DoorRadianceProperties(_PropertiesBase):
+    """Radiance Properties for Honeybee Door."""
+
+    type: constr(regex='^DoorRadianceProperties$') = 'DoorRadianceProperties'
+
+
+class FaceRadiancePropertiesAbridged(_PropertiesBaseAbridged):
+    """Radiance Properties for Honeybee Face Abridged."""
+
+    type: constr(regex='^FaceRadiancePropertiesAbridged$') = \
+        'FaceRadiancePropertiesAbridged'
+
+
+class FaceRadianceProperties(_PropertiesBase):
+    """Radiance Properties for Honeybee Face."""
+
+    type: constr(regex='^FaceRadianceProperties$') = 'FaceRadianceProperties'
+
+
+class ShadeRadiancePropertiesAbridged(_PropertiesBaseAbridged):
+    """Radiance Properties for Honeybee Shade Abridged."""
+
+    type: constr(regex='^ShadeRadiancePropertiesAbridged$') = \
+        'ShadeRadiancePropertiesAbridged'
+
+
+class ShadeRadianceProperties(_PropertiesBase):
+    """Radiance Properties for Honeybee Shade."""
+
+    type: constr(regex='^ShadeRadianceProperties$') = 'ShadeRadianceProperties'
+
+
+class ModelRadianceProperties(IDdBaseModel):
+    """Radiance Properties for Honeybee Model."""
+
+    type: constr(regex='^ModelRadianceProperties$') = 'ModelRadianceProperties'
+
+
+    # TODO: Clarification: does there need to be an abridged version of all these lists?
+    modifiers: List[
+        ModifierBase
+    ] = Field(
```

Comment:
This will not be valid as far as I know. You need the exact object to match here.

Reply:
@mostaphaRoudsari I must be misunderstanding your earlier comment:

> Create a base class and subclass all the other Radiance modifiers from the baseclass for now similar to how they are implemented in honeybee-radiance. Otherwise I can see this to be a problem as soon as people start using the schema.

`ModifierBase` here is the base class that I use to create all the modifier subclasses. Is that not what you meant?

Reply:
Sorry. I should have written a longer note. I understand that you are using the base class but OpenAPI schema check doesn't work like Python's `isinstance`. The objects should exactly match be valid.

When I suggested to create a list I mean to create a list of all the subclasses. An since there are several subclasses that you don't have a schema for then we can create a dummy subclass like how they have been implemented in honeybee-radiance Python implementation.

Here is an example for BRTDfunc:

```python 

class BRTDfunc(ModifierBase):
    type: str = 'BRTDfunc'   # or the correct implementation with constant

```

and now we can add this to the list of supported modifiers.

Reply:
I made a comment about this earlier. This is why we need to use Unions of all the Modifiers that we currently support.

Reply:
I'll create a constant at the top of the file with the union of all the modifiers.